src/init.cpp

```diff
@@ -273,7 +274,7 @@ void Shutdown()
  */
 static void HandleSIGTERM(int)
 {
-    fRequestShutdown = true;
+    StartShutdown();
```

Comment:
This is not a good idea, UNIX signal handlers are extremely limited in what they can do. Basically:
- Set atomic flags
- Send to/receive from pipe
This is why the polling was kept in #10125. It is there *in addition* to notification-based user initiated shutdown.